### ISSUE TITLE
Details about training part

### ISSUE THREAD

**Issue by ukaukaaaa** [sail-sg/EditAnything#31]
Hi, thanks for this great project. I have some questions about the training part.

1. From data_build.py, we know that there is one txt file and a bunch of json files, I wonder if you can release an example of both txt and json files.
2. Is the updated checkpoint model trained by current released training code? If not, would you mind to release the newest training code?
3. If possible, can you complement more training details in README?

**Comment by gasvn** [comment]
1.The txt and json files can be refer to https://github.com/lllyasviel/ControlNet
2.The latest verision is trained with laion dataset, so the training code is a little bit differet, we will release it soon.
3.More training details will be added.

**Comment by ukaukaaaa** [comment]
> 1.The txt and json files can be refer to https://github.com/lllyasviel/ControlNet 2.The latest verision is trained with laion dataset, so the training code is a little bit differet, we will release it soon. 3.More training details will be added.

Thanks for your reply,

Since in this project, mask image is taken as an extra source input to the model, so will that change the training part, such as the json file?

Also, if I want to add one more image as a guidance such as clothing item, is it possible to incorporate this in the training and inference framework?

**Comment by alelordelo** [comment]
@ukaukaaaa @gasvn Im also interested on this.

A training guide to train JSON COCO (with annotation and segmented masks) would be super helpful!

**Comment by gasvn** [comment]
> > 1.The txt and json files can be refer to https://github.com/lllyasviel/ControlNet 2.The latest verision is trained with laion dataset, so the training code is a little bit differet, we will release it soon. 3.More training details will be added.
> 
> Thanks for your reply,
> 
> Since in this project, mask image is taken as an extra source input to the model, so will that change the training part, such as the json file?
> 
> Also, if I want to add one more image as a guidance such as clothing item, is it possible to incorporate this in the training and inference framework?

Sorry I missed your comments. I am not sure about the exact form of how to use "one more image" as the guidance. Maybe the biggest problem here is the training data.  Also, you can check our latest version, where I achieve a training free drag item from one image into another based on reference scheme. But of course, there are some failure cases as no training is evolved. With training data, I think the current pipeline would be improved.

**Comment by gasvn** [comment]
> @ukaukaaaa @gasvn Im also interested on this.
> 
> A training guide to train JSON COCO (with annotation and segmented masks) would be super helpful!

The training data is the same to the basic controlnet. As long as you can construct a dataloader that produce the data with     jpg = item['jpg']
    txt = item['txt']
    hint = item['hint']， you can use this training code.

**Comment by alelordelo** [comment]
Thanks. Loading the data is fine, the challenge is how to do multi prompt per segmentation mask?

**Comment by leshhh-aneee** [comment]
Hi, you mentioned that the training dataset is exactly the same as Controlnet, however controlnet dataset is using images as their source and target, whereas the code in 'sam_dataset.py' is using source as a json file and target as an image. if source is a json file, what should be included in the annotations? 

what is annotations , height and width suppose to mean? please help T.T
